### PR TITLE
Support LCM in ControlNet and Adapter pipelines.

### PR DESCRIPTION
**Pull request by patil-suraj**
# What does this PR do?

This PR adds support for LCM in:
- `StableDiffusionControlNetPipeline`
- `StableDiffusionXLControlNetPipeline`
- `StableDiffusionAdapterPipeline`
- `StableDiffusionXLAdapterPipeline`

Also added fast tests for LCMs with these pipelines.

### CODE REVIEW

File: tests/pipelines/stable_diffusion/test_stable_diffusion_adapter.py [hunk 1]
```diff
@@ -264,13 +266,13 @@ def test_inference_batch_single_identical(self):
     @parameterized.expand(
         [
             # (dim=264) The internal feature map will be 33x33 after initial pixel unshuffling (downscaled x8).
-            ((4 * 8 + 1) * 8),
+            (((4 * 8 + 1) * 8),),
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion/test_stable_diffusion_adapter.py:269]: change on purpose?

**Review by patrickvonplaten — APPROVED** [review]
Nice!

File: tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl_adapter.py [hunk 2]
```diff
@@ -309,9 +311,9 @@ def test_stable_diffusion_adapter_default_case(self):
     @parameterized.expand(
         [
             # (dim=144) The internal feature map will be 9x9 after initial pixel unshuffling (downscaled x16).
-            ((4 * 2 + 1) * 16),
+            (((4 * 2 + 1) * 16),),
             # (dim=160) The internal feature map will be 5x5 after the first T2I down block (downscaled x32).
-            ((4 * 1 + 1) * 32),
+            (((4 * 1 + 1) * 32),),
```
[the remark this answers is not in the document]
Comment (patil-suraj) [reply tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl_adapter.py:316]: Yes, without this, the parametrized tests were erroing.

**Comment by patil-suraj** [comment]
The failing test is unrelated. Merging now.

**Pull request merged**